### PR TITLE
Enable fileNameGenerator byType to read Javscript files by mimeType. Curr…

### PR DESCRIPTION
**Pull request by ykhandor**
Enable fileNameGenerator "byType" to read Javscript files by mimeType. Currently code only enables CSS and HTML files by mimeType.

### CODE REVIEW

File: lib/config/resource-type-by-ext.js [hunk 1]
```diff
@@ -4,5 +4,7 @@ var types = require('./resource-types');
 module.exports = {
 	'.html': types.html,
 	'.htm': types.html,
-	'.css': types.css
+	'.css': types.css,
+	'.js': types.js,
+	'.min.js': types.js
```
Comment (s0ph1e) [inline lib/config/resource-type-by-ext.js]: I believe `.min.js` is not needed here, such files will have extension`.js`

Comment (ykhandor) [reply lib/config/resource-type-by-ext.js]: Makes sense..will get rid of this one. I was not sure, I added it to be double sure and cover all bases.

Comment (ykhandor) [reply lib/config/resource-type-by-ext.js]: Also, I will add the test for .js files and push the code up. You should have a new commit/PR to approve soon!

Comment (ykhandor) [reply lib/config/resource-type-by-ext.js]: @s0ph1e : The updated code has been pushed and the new commit is in and ready for approval.

**Comment by s0ph1e** [comment]
Thanks @ykhandor 
I will publish this changes in next release 3.3.4 today

**Pull request merged**